### PR TITLE
CB-18752 Allow multiAz when StackType is TEMPLATE

### PR DESCRIPTION
**Pull request by balassai**
See detailed description in the commit message.

### CODE REVIEW

**Review by urbandan — APPROVED** [review]

**Comment by balassai** [comment]
retest this please

**Review by benyoka — APPROVED** [review]

File: template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kafka/KafkaMultiAzConfigProvider.java [hunk 1]
```diff
@@ -54,7 +54,8 @@ public String getServiceType() {
 
     @Override
     public boolean isConfigurationNeeded(CmTemplateProcessor cmTemplateProcessor, TemplatePreparationObject source) {
-        return (StackType.DATALAKE.equals(source.getStackType()) || StackType.WORKLOAD.equals(source.getStackType()))
+        return (StackType.DATALAKE.equals(source.getStackType()) || StackType.WORKLOAD.equals(source.getStackType()) ||
+            StackType.TEMPLATE.equals(source.getStackType()))
```
Comment (lajosrodek) [inline template-manager-cmtemplate/src/main/java/com/sequenceiq/cloudbreak/cmtemplate/configproviders/kafka/KafkaMultiAzConfigProvider.java:58]: Sorry but this does not make sense. The stack of clusters cannot be of `StackType.TEMPLATE`. The latter represents a customer cluster definition, that is called a "cluster template" in the CB repo for historical reasons (Ambari times). The stack is either `DATALAKE` or `WORKLOAD` (for DH clusters), even if they are launched from a custom cluster definition (that is internally stored as a stack of `StackType.TEMPLATE`). Adding this extra condition will not solve your issues.

**Review by lajosrodek — CHANGES_REQUESTED** [review]

**Comment by lajosrodek** [comment]
CC @biharitomi

**Comment by balassai** [comment]
Closing.

**Pull request closed without merging**